### ISSUE TITLE
xmlgen should ignore non-supported XML w/ a warning

### ISSUE THREAD

**Issue by zeenix** [dbus2/zbus#256]
Creating a separate issue for the simpler/bug part of #255. As [discussed there](https://gitlab.freedesktop.org/dbus/zbus/-/issues/255#note_1318643), we should ignore any unexpected XML with a warning on the standard error output, instead of just completely giving up.

**Comment by kb-1000** [comment]
As far as I've seen so far, the convention with XML usually tends to be silently ignoring elements and attributes from unknown namespaces, which are usually used for ignorable vendor/etc. extensions, see for example JAXB's XML schema binding extensions that no one other than JAXB needs or reads, or some of LibreOffice's OpenDocument extensions. A warning is probably not necessary or desirable, as long as the known elements and attributes are okay, an info/debug-level message controlled by a verbose mode would make more sense.

**Comment by zeenix** [comment]
> A warning is probably not necessary or desirable

I disagree but that's not important. The important bit is to ignore unknown elements. Do you think you can contribute this?

**Comment by kira-bruneau** [comment]
It looks like won't be straightforward to fix. quick-xml doesn't distinguish between namespaces when using serde deserialization, so zbus_xml will have to be refactored to use something like [NsReader](https://docs.rs/quick-xml/latest/quick_xml/reader/struct.NsReader.html) or [xmlserde](https://lib.rs/crates/xmlserde).

**Comment by kira-bruneau** [comment]
Hmm, actually an easy workaround would be to create a fork of quick-xml that ignores all properties under a namespace like described in: https://github.com/tafia/quick-xml/issues/547#issuecomment-1406711576

**Comment by zeenix** [comment]
> It looks like this won't be straightforward to fix.

Yeah, definitely not. :(

> quick-xml doesn't distinguish between namespaces when using serde deserialization: [tafia/quick-xml#757](https://github.com/tafia/quick-xml/issues/757),

More like https://github.com/tafia/quick-xml/issues/218 .

> so zbus_xml will have to be refactored to use something like [NsReader](https://docs.rs/quick-xml/latest/quick_xml/reader/struct.NsReader.html) or [xmlserde](https://lib.rs/crates/xmlserde)

Yeah. Also `xml` crate might not be a bad idea to try. It's not serde and we'll need more manual deserialization code but you can't have it all I guess.

There is also the option of people needing this, to work on https://github.com/tafia/quick-xml/issues/218 getting fixed. I know that's a tall order but at least [there is already some WIP work on that done](https://github.com/tafia/quick-xml/pull/466).
